src/index.spec.ts

```diff
@@ -1,12 +1,180 @@
+    expect(toLowerCase('THIS IS A TEST')).toEqual('this is a test');
+  });
+  
+  test('Return lowercase if we pass lowercase string', () => {
+    expect(toLowerCase('this is a test')).toEqual('this is a test')
+  })
+});
+
+describe('removeDuplicatesFromArray', () => {
+  const duplicateItemsArray = ['@', '@', 2, 2];
+  const uniqueItemsArray = ['this', 'is', 'a', 'test', 1, 2, 3];
+  const singleItemArray = [1];
+  const emptyArray = [];
+  const hashValue = {key: 'value'};
+  const nullValue = null
+
+  test('Return removed items array if we pass an array with duplicate items', () => {
+    expect(removeDuplicatesFromArray(duplicateItemsArray)).toEqual(['@', 2]);
+  });
+
+  test('Return same array if we pass an array with unique items', () => {
+    expect(removeDuplicatesFromArray(uniqueItemsArray)).toEqual(uniqueItemsArray);
+  });
+
+  test('Return empty array if we pass an empty array', () => {
+    expect(removeDuplicatesFromArray(emptyArray)).toEqual(emptyArray);
+  });
+  
+  test('Return same array if we pass an array with a single item', () => {
+    expect(removeDuplicatesFromArray(singleItemArray)).toEqual(singleItemArray);
+  });
+
+  test('Throw an error when pass invalid value', () => {
+    function passEmptyArray(){
+      removeDuplicatesFromArray(hashValue as any);
+    };
+    expect(passEmptyArray).toThrow(Error);
+  });
+
+  test('Throw an error when pass invalid value', () => {
+    function passEmptyArray(){
+      removeDuplicatesFromArray(nullValue as any);
+    };
+    expect(passEmptyArray).toThrow(Error);
+  });
 });
+
+describe('createProduct', () => {
+  const validProduct = {
+    name: 'Eggs',
+    tags: ['test tag'],
+    description: 'Description about good eggs',
+    price: 15.30,
+  };
+  const invalidProduct = {
+    id: 'bad id',
+    name: '1234',
+    tags: ['dairy'],
+    description: 'Raw organic brown eggs in a basket',
+    price: 28.1,
+  };
+
+  test('Return a valid product if we pass a valid schema', () => {
+    expect(createProduct(validProduct)).toEqual({
+      id: expect.any(Number),
+      name: validProduct.name,
+      tags: validProduct.tags,
+      description: validProduct.description,
+      price: validProduct.price,
+    });
+  });
+
+  test('Throw an error if we pass an invalid product', () => {
+    expect(() => createProduct(invalidProduct as any)).toThrow(Error);
+  });
+});
+
+describe('createRandomProduct', () => {
+  test('Create a product with a correct user role', () => {
+    expect(createRandomProduct('clark@kent.com')).toEqual({
+      id: expect.any(Number),
+      name: expect.any(String),
+      description: expect.any(String),
+      tags: expect.any(Array),
+      price: expect.any(String),
+    });
+  });
+
+  test('Throws an error with an incorrect user role', () => {
+    expect(() => {
+      createRandomProduct('diana@themyscira.com');
+    }).toThrowError(Error);
+  });
+});
+
+jest.mock('node-fetch', () => {
+  return jest.fn();
+});
```

Comment:
Mocked libraries should be defined at the top of the file, before importing any library